app/workloadSummaryReport/services/workloadSummaryActions.js

```diff
@@ -0,0 +1,562 @@
+				});
+			},
+			_performCalculations: function () {
+				this._isInitialFetchComplete();
+
+				if (WorkloadSummaryReducers._state.calculations.isInitialFetchComplete && WorkloadSummaryReducers._state.calculations.censusDataFetchBegun == false) {
+					this._getEnrollmentData();
+					this._getEnrollmentData(true);
+				}
+
+				if (WorkloadSummaryReducers._state.calculations.isInitialFetchComplete) {
+					this._calculateView();
+				}
+			},
+			_isInitialFetchComplete: function () {
+				var sectionGroups = WorkloadSummaryReducers._state.sectionGroups;
+				var courses = WorkloadSummaryReducers._state.courses;
+				var teachingAssignments = WorkloadSummaryReducers._state.teachingAssignments;
+				var instructors = WorkloadSummaryReducers._state.instructors;
+				var instructorTypes = WorkloadSummaryReducers._state.instructorTypes;
+				var users = WorkloadSummaryReducers._state.users;
+				var userRoles = WorkloadSummaryReducers._state.userRoles;
+				var sections = WorkloadSummaryReducers._state.sections;
+
+				if (sectionGroups && courses && teachingAssignments && instructors && instructorTypes && users && userRoles && sections) {
+					WorkloadSummaryReducers.reduce({
+						type: ActionTypes.INITIAL_FETCH_COMPLETE,
+						payload: {
+							isInitialFetchComplete: true
+						}
+					});
+				}
+			},
+			_calculateView: function () {
+				var _self = this;
+
+				var sectionGroups = WorkloadSummaryReducers._state.sectionGroups;
+				var courses = WorkloadSummaryReducers._state.courses;
+				var teachingAssignments = WorkloadSummaryReducers._state.teachingAssignments;
+				var instructors = WorkloadSummaryReducers._state.instructors;
+				var instructorTypes = WorkloadSummaryReducers._state.instructorTypes;
+				var users = WorkloadSummaryReducers._state.users;
+				var userRoles = WorkloadSummaryReducers._state.userRoles;
+
+				var calculatedView = {
+					instructorTypeIds: [],
+					byInstructorType: {}
+				};
+
+				instructors.ids.forEach(function(instructorId) {
+					var instructor = instructors.list[instructorId];
+					var instructorTypeId = _self._getInstructorTypeId(instructor);
+
+					if (calculatedView.instructorTypeIds.indexOf(instructorTypeId) == -1) {
+						calculatedView.instructorTypeIds.push(instructorTypeId);
+						calculatedView.byInstructorType[instructorTypeId] = [];
+					}
+
+					instructor.assignments = [];
+					instructor.totals = {
+						units: 0,
+						studentCreditHours: 0,
+						enrollment: 0,
+						previousEnrollment: 0,
+						assignmentCount: 0
+					};
+
+					var instructorAssignments = _self._getInstructorAssignments(instructorId, teachingAssignments);
+
+					instructorAssignments.forEach(function(teachingAssignment) {
+						var assignment = {};
+
+						var termCode = teachingAssignment.termCode;
+
+						assignment.term = TermService.getTermName(termCode);
+
+						assignment.description = null;
+
+						if (teachingAssignment.buyout) {
```

Comment:
Can you establish a teaching assignment service that translates these values into text so this is DRYed up in our codebase?